File: frontend/test-api.js
const axios = require('axios');
(async () => {
    let res = await axios.post('http://localhost:8000/api/v1/auth/login/', {email: "test_owner@loyallia.com", password: "123456"});
    let token = res.data.access_token;
    let list = await axios.get('http://localhost:8000/api/v1/customers/', {headers: {Authorization: `Bearer ${token}`}});
    let id = list.data.customers[0].id;
    try {
        let cust = await axios.get(`http://localhost:8000/api/v1/customers/${id}/`, {headers: {Authorization: `Bearer ${token}`}});
        console.log("Customer Fetch Success!");
    } catch(e) {
        console.error("Customer Fetch Failed:", e.response.status, e.response.data);
    }
})();

File: backend/test-api.py
import sys

import requests

res = requests.post('http://localhost:8000/api/v1/auth/login/', json={"email": "test_owner@loyallia.com", "password": "123456"})
token = res.json()["access_token"]

list_res = requests.get('http://localhost:8000/api/v1/customers/', headers={"Authorization": f"Bearer {token}"})
try:
    first_id = list_res.json()["customers"][0]["id"]
except Exception:
    print("NO CUSTOMERS:", list_res.text)
    sys.exit(1)

cust_res = requests.get(f'http://localhost:8000/api/v1/customers/{first_id}/', headers={"Authorization": f"Bearer {token}"})
if cust_res.status_code == 200:
    print("Customer Fetch Success:", cust_res.json()["first_name"])
else:
    print("Customer Fetch Failed:", cust_res.status_code, cust_res.text)
